sdk/src/nonce_state.rs

```diff
@@ -0,0 +1,610 @@
+use crate::{
```

Comment:
Does `nonce_state.rs` need to be in the SDK? Why can't `runtime` add `solana-nonce-program` as a direct dependency?

Reply:
It doesn't.  The only thing linking`nonce_instruction.rs` back to `nonce_state.rs` is the state size convenience function, which I can duplicate.

Can you elaborate on "direct dependency"?   I tried to follow `system_program`